### DISCUSSION TITLE
Password Confirm Failed

### DISCUSSION THREAD

**Discussion by dev-techguy** [livewire/livewire#6506] [7. Bugs (archived)]
### Livewire version

v3

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Create UserForm`    #[Rule(['required', 'string', new Password(8)])]
    public string $password = '';

    #[Rule('required', message: 'The password confirm field is required.')]
    #[Rule('same:password', message: 'The password field confirmation does not match.')]
    public string $passwordConfirmation = '';`

2. Use the UserForm on the Component
3. Add the following on blade`                      <div class="mb-3">
                                <div class="form-floating">
                                    <input type="password"
                                           class="form-control @error('form.password') is-invalid @enderror"
                                           id="form.password" required placeholder="xxxxxx"
                                           wire:model.live="form.password">
                                    <label for="form.password">Password</label>
                                    @error('form.password')
                                    <span class="invalid-feedback" role="alert">
                                        <strong>{{ $message }}</strong>
                                    </span>
                                    @enderror
                                </div>
                            </div>
                            <div class="mb-3">
                                <div class="form-floating">
                                    <input type="password"
                                           class="form-control @error('form.passwordConfirmation') is-invalid @enderror"
                                           name="form.passwordConfirmation"
                                           id="form.passwordConfirmation" required placeholder="xxxxxx"
                                           wire:model.live="form.passwordConfirmation">
                                    <label for="form.passwordConfirmation">Confirm Password</label>
                                    @error('form.passwordConfirmation')
                                    <span class="invalid-feedback" role="alert">
                                        <strong>{{ $message }}</strong>
                                    </span>
                                    @enderror
                                </div>
                            </div>`
4. Try Inputting password and confirm
5. Error
![Screenshot from 2023-08-24 22-53-30](https://github.com/livewire/livewire/assets/36882600/cae97dc6-04c1-4142-a2aa-bd26b5fe811f)

### Additional information

After inputting the same password still get a password don't match issue.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by tahirafridi** [comment]
This is what I faced also, then I used the v2 way of validation with validate() method

**Comment by Shakti962** [comment]
To resolve this issue, just define both `$password` and `$password_confirmation` properties.
